### ISSUE TITLE
New postfix filter mode to match "too many errors"

### ISSUE THREAD

**Issue by K-os** [fail2ban/fail2ban#2439]
By default postfix disconnects an SMTP client after 20 errors with the message "too many errors" (postconf: `smtpd_hard_error_limit`), but allows the client to reconnect immediately. Here is a suggestion for a rule to ban a client that has been disconnected by postfix due to too many errors.

```diff
diff --git a/config/filter.d/postfix.conf b/config/filter.d/postfix.conf
index dfc68b5a..3bb6b2be 100644
--- a/config/filter.d/postfix.conf
+++ b/config/filter.d/postfix.conf
@@ -48,6 +48,8 @@ mdpr-aggressive = (?:%(mdpr-auth)s|%(mdpr-normal)s|%(mdpr-ddos)s)
 mdre-aggressive = %(mdre-auth2)s
                   %(mdre-normal)s
 
+mdpr-many_errors = too many errors
+mdre-many_errors = ^after \S+ from [^[]*\[<HOST>\]%(_port)s$
 
 
 failregex = <mdre-<mode>>
```

This is an example for a corresponding log-line:
```
Jun  3 06:25:43 server2 postfix/smtpd[29306]: too many errors after RCPT from srv-eml.info[185.222.211.11]
```

Use it like this:
```ini
[postfix-too-many-errors]
filter   = postfix[mode=many_errors]
port     = smtp,465,submission
logpath  = %(postfix_log)s
backend  = %(postfix_backend)s
maxretry = 1
```

**Comment by sebres** [comment]
And what all those errors are?
The question is why to react firstly on the summary message (after 20 errors), instead of on each messages?
Because the simplest hack to avoid your jail would be - "do maximal 19 errors per session only".

**Comment by K-os** [comment]
There are many errors that are not logged by postfix, e.g. invalid SMTP commands, or commands that are not valid at the current protocol state.

**Comment by sebres** [comment]
OK, possibly it may be useful...  agree.

So why not to add it simultaneously to normal (and/or aggressive) modes too... (just because an additional jail produces always unneeded load extra).
Like this:
```diff
-mdpr-normal = (?:(?:\w+): reject:|improper command pipelining after \S+)
+mdpr-normal = (?:(?:\w+): reject:|(?:improper command pipelining|too many errors) after \S+)
 mdre-normal = ^RCPT from [^[]*\[<HOST>\]%(_port)s: 55[04] 5\.7\.1\s
 ...
               ^from [^[]*\[<HOST>\]%(_port)s:?

+mdpr-errors = too many errors after \S+
+mdre-errors = ^from [^[]*\[<HOST>\]%(_port)s$

**Comment by K-os** [comment]
I added it to a separate jail, because I want the ban to be triggered on the first „too many errors“ message, which is probably too aggressive for the normal mode.

**Comment by sebres** [comment]
Well, basically normal mode used now also for sundry non authentication issues (there is `auth` mode for that), so it would trigger by various client reject messages, like no host, no domain or no reverse hostname as well as several protocol deviations for any reason (like helo rejected, etc).
And `maxretry` is 3 or even higher normally, thus in my opinion it is a good candidate for enhancement like this.
And we can additionally retain your initial idea with new mode (for the separate jail) too.

**Comment by sebres** [comment]
Implemented in e5ae113215174e6814284d208725913c85991c87 as new mode `errors` (for example to use in new jail with `maxrety = 1`) as well as included to modes `normal`, `more` (`extra` and `aggressive`) for default postfix jail.

**Comment by K-os** [comment]
Ok cool, thanks! The diff looks good to me, but I haven't tested it yet.
